test/unit/node/util.test.ts

```diff
@@ -447,6 +447,33 @@ describe("onLine", () => {
 
     expect(await received).toEqual(expected)
   })
+
+  describe("used with a process missing stdout ", () => {
+    it("should throw an error", async () => {
+      // Initialize a process that does not have stdout.
+      // "If the child was spawned with stdio set to anything
+      // other than 'pipe', then subprocess.stdout will be null."
+      // Source: https://stackoverflow.com/a/46024006/3015595
+      // Other source: https://nodejs.org/api/child_process.html#child_process_subprocess_stdout
+      // NOTE@jsjoeio - I'm not sure if this actually happens though
+      // which is why I have to set proc.stdout = null
+      // a couple lines below.
+      const proc = cp.spawn("node", [], {
+        stdio: [null],
+      })
+      const mockCallback = jest.fn()
+
+      // @ts-expect-error TypeScript knows proc.stdout should return a Readable
+      // We overwrite it so that we can get testing coverage for the scenario
+      // in onLine that would throw an error.
+      proc.stdout = null
```

Comment:
According to the docs, it doesn't seem like I should have to do this but no matter what values I pass in the array to `stdio` it seems like `proc.stdout` is always set? I would love a second opinion here cc @vapurrmaid @bryphe-coder

Reply:
I think you might need different settings for `cp.spawn` - checking

Reply:
If I use the `stdio` flag `'ignore'`, it seems to work as you expect (in order to exercise the code under test):

```
const proc = cp.spawn("node", [], {
   stdio: 'ignore'
})
```

Reply:
At least, tested this out in the NodeJS console: 
![image](https://user-images.githubusercontent.com/88213859/142922783-5b6cacb4-1b77-4969-a94f-45230916c4e7.png)

Reply:
Here is a quote from the [v14 docs](https://nodejs.org/docs/latest-v14.x/api/child_process.html#child_process_subprocess_stdout):
> The `subprocess.stdout` property can be `null` if the child process could not be successfully spawned.

Seems like that's what this person was testing for. I'm not sure how to simulate this though (also wondering if it's worth the effort/time since my current implementation simulates it in _one way_ (albeit hacky).

Reply:
Some useful docs around the various settings for `options.stdio` here: https://nodejs.org/api/child_process.html#optionsstdio

Reply:
Oh wow! Nice work 🙌 Okay I'll try that locally and see if it works for me as well.

Reply:
this one is better than my suggestion - glad it worked!